lib/stanford-mods/name.rb

```diff
@@ -53,10 +53,13 @@ def additional_authors_w_dates
       def non_collector_person_authors
         result = []
         @mods_ng_xml.personal_name.map do |n|
-          r = n.role
-          unless (r.authority.include?('marcrelator') && r.value.include?('Collector')) ||
-                  r.roleTerm.valueURI.first == COLLECTOR_ROLE_URI
-            result << n.display_value_w_date
+          unless n.role.size == 0
```

Comment:
We could reduce a bit of the complexity of this loop if we do something like `next if n.role.size.zero?` instead of wrapping the logic in a guard clause.